ktlint-core/src/main/kotlin/com/pinterest/ktlint/core/internal/ThreadSafeEditorConfigCache.kt

```diff
@@ -0,0 +1,35 @@
+package com.pinterest.ktlint.core.internal
+
+import java.util.concurrent.locks.ReentrantReadWriteLock
+import kotlin.concurrent.read
+import kotlin.concurrent.write
+import org.ec4j.core.Cache
+import org.ec4j.core.EditorConfigLoader
+import org.ec4j.core.Resource
+import org.ec4j.core.model.EditorConfig
+
+/**
+ * In-memory [Cache] implementation that could be safely accessed from any [Thread].
+ */
+internal class ThreadSafeEditorConfigCache : Cache {
```

Comment:
not sure if it's worth to test this class?

Reply:
It is pretty straightforward, but I will add some tests :+1:

Reply:
After some thoughts I've decided to not write tests for this class. Class itself is pretty straightforward, though writing non-flaky multithreading tests are hard and not worth, in this particular case, an effort.